In [2]:
import numpy as np
import pandas as pd

In [3]:
#montar o drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
dados = pd.read_csv('/content/drive/MyDrive/Microdados-enem2022/enem_2022_tratado.csv',
                    sep=',',encoding='iso-8859-1')
# encoding: codificação de caracteres, normalmente utiliza-se o iso-8859-1, utf-8, latin-1)

In [5]:
pd.options.display.max_columns = 44

In [6]:
dados.head()

NU_INSCRICAO  NU_ANO  TP_FAIXA_ETARIA TP_SEXO  TP_ESTADO_CIVIL  \
0  210057943671    2022               14       M                2   
1  210057516120    2022               14       M                2   
2  210057280536    2022                5       F                1   
3  210055724397    2022                6       M                1   
4  210055097896    2022                4       M                0   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  TP_ESCOLA  \
0            2                 1                1                2          1   
1            1                 1                1               16          1   
2            2                 1                1                2          1   
3            3                 1                1                2          1   
4            3                 1                1                1          1   

   TP_ENSINO  IN_TREINEIRO  TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  \
0        NaN             0                 NaN                   Brasília   
1        NaN             0                 NaN                   Brasília   
2        NaN             0                 NaN  Presidente Tancredo Neves   
3        NaN             0                 NaN                  Cariacica   
4        NaN             0                 NaN                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0         NaN         NaN         NaN         NaN          0   
1         NaN         NaN         NaN         NaN          0   
2       421.1       546.0       498.8       565.3          1   
3       490.7       388.6       357.8       416.0          1   
4         NaN         NaN         NaN         NaN          0   

   TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0                NaN              NaN  
1                NaN              NaN  
2                1.0            760.0  
3                1.0            320.0  
4                NaN              NaN

#Corrigindo erros nas notas

In [7]:
#loc: localizar
dados.loc[:,'NU_NOTA_CN'] /= 10

In [8]:
#head() é um metodo, é possivel passar parametros dentro do metodo
dados.head(12)

NU_INSCRICAO  NU_ANO  TP_FAIXA_ETARIA TP_SEXO  TP_ESTADO_CIVIL  \
0   210057943671    2022               14       M                2   
1   210057516120    2022               14       M                2   
2   210057280536    2022                5       F                1   
3   210055724397    2022                6       M                1   
4   210055097896    2022                4       M                0   
5   210057850231    2022                5       F                1   
6   210056305481    2022                4       M                1   
7   210055778089    2022                6       F                1   
8   210057252239    2022                8       F                1   
9   210055461467    2022                4       F                1   
10  210055665589    2022                4       F                1   
11  210054972570    2022                2       F                1   

    TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  \
0             2                 1                1                2   
1             1                 1                1               16   
2             2                 1                1                2   
3             3                 1                1                2   
4             3                 1                1                1   
5             3                 1                1                2   
6             2                 1                1                2   
7             2                 1                1                0   
8             3                 1                1                0   
9             3                 1                1                1   
10            3                 1                1                2   
11            1                 1                2                0   

    TP_ESCOLA  TP_ENSINO  IN_TREINEIRO  TP_LOCALIZACAO_ESC  \
0           1        NaN             0                 NaN   
1           1        NaN             0                 NaN   
2           1        NaN             0                 NaN   
3           1        NaN             0                 NaN   
4           1        NaN             0                 NaN   
5           1        NaN             0                 NaN   
6           1        NaN             0                 NaN   
7           1        NaN             0                 NaN   
8           1        NaN             0                 NaN   
9           1        NaN             0                 NaN   
10          1        NaN             0                 NaN   
11          3        1.0             0                 NaN   

           NO_MUNICIPIO_PROVA  CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  \
0                    Brasília           53          DF               0   
1                    Brasília           53          DF               0   
2   Presidente Tancredo Neves           29          BA               1   
3                   Cariacica           32          ES               1   
4                 Salinópolis           15          PA               0   
5              Rio de Janeiro           33          RJ               0   
6                      Recife           26          PE               0   
7                 São Gonçalo           33          RJ               1   
8                    Amargosa           29          BA               0   
9                   Arcoverde           26          PE               1   
10                      Patos           25          PB               0   
11    Nossa Senhora da Glória           28          SE               1   

    TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  \
0                0               0               0          NaN          NaN   
1                0               0               0          NaN          NaN   
2                1               1               1       1087.0       1056.0   
3                1               1               1       1088.0       1058.0   

In [9]:
#corrigir todas as colunas de nota
dados.loc[:,'NU_NOTA_CH'] /= 10
dados.loc[:,'NU_NOTA_LC'] /= 10
dados.loc[:,'NU_NOTA_MT'] /= 10

#Renomeando Variáveis (colunas)

In [10]:
#Renomear variaveis
dados1 = dados.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [11]:
dados1.head()

NU_INSCRICAO  NU_ANO  TP_FAIXA_ETARIA TP_SEXO  TP_ESTADO_CIVIL  \
0  210057943671    2022               14       M                2   
1  210057516120    2022               14       M                2   
2  210057280536    2022                5       F                1   
3  210055724397    2022                6       M                1   
4  210055097896    2022                4       M                0   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  TP_ESCOLA  \
0            2                 1                1                2          1   
1            1                 1                1               16          1   
2            2                 1                1                2          1   
3            3                 1                1                2          1   
4            3                 1                1                1          1   

   TP_ENSINO  IN_TREINEIRO  TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  \
0        NaN             0                 NaN                   Brasília   
1        NaN             0                 NaN                   Brasília   
2        NaN             0                 NaN  Presidente Tancredo Neves   
3        NaN             0                 NaN                  Cariacica   
4        NaN             0                 NaN                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0         NaN         NaN         NaN         NaN          0   
1         NaN         NaN         NaN         NaN          0   
2       42.11       54.60       49.88       56.53          1   
3       49.07       38.86       35.78       41.60          1   
4         NaN         NaN         NaN         NaN          0   

   TP_STATUS_REDACAO  NOTA_REDACAO  
0                NaN           NaN  
1                NaN           NaN  
2                1.0         760.0  
3                1.0         320.0  
4                NaN           NaN

In [12]:
dados1 = dados.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [13]:
dados1.head()

NU_INSCRICAO  NU_ANO  TP_FAIXA_ETARIA TP_SEXO  TP_ESTADO_CIVIL  \
0  210057943671    2022               14       M                2   
1  210057516120    2022               14       M                2   
2  210057280536    2022                5       F                1   
3  210055724397    2022                6       M                1   
4  210055097896    2022                4       M                0   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  TP_ESCOLA  \
0            2                 1                1                2          1   
1            1                 1                1               16          1   
2            2                 1                1                2          1   
3            3                 1                1                2          1   
4            3                 1                1                1          1   

   TP_ENSINO  IN_TREINEIRO  TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  \
0        NaN             0                 NaN                   Brasília   
1        NaN             0                 NaN                   Brasília   
2        NaN             0                 NaN  Presidente Tancredo Neves   
3        NaN             0                 NaN                  Cariacica   
4        NaN             0                 NaN                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      NaN      NaN      NaN      NaN          0                NaN   
1      NaN      NaN      NaN      NaN          0                NaN   
2    42.11    54.60    49.88    56.53          1                1.0   
3    49.07    38.86    35.78    41.60          1                1.0   
4      NaN      NaN      NaN      NaN          0                NaN   

   NU_NOTA_REDACAO  
0              NaN  
1              NaN  
2            760.0  
3            320.0  
4              NaN

In [14]:
dados1 = dados.rename(columns={'TP_SEXO': 'SEXO','TP_COR_RACA':'RACA','TP_ESTADO_CIVIL':'ESTADO_CIVIL',
                             'TP_FAIXA_ETARIA':'IDADE','TP_NACIONALIDADE':'NACIONALIDADE','TP_ESCOLA':'ESCOLA' })

In [15]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO  ESTADO_CIVIL  RACA  NACIONALIDADE  \
0  210057943671    2022     14    M             2     2              1   
1  210057516120    2022     14    M             2     1              1   
2  210057280536    2022      5    F             1     2              1   
3  210055724397    2022      6    M             1     3              1   
4  210055097896    2022      4    M             0     3              1   

   TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  ESCOLA  TP_ENSINO  IN_TREINEIRO  \
0                1                2       1        NaN             0   
1                1               16       1        NaN             0   
2                1                2       1        NaN             0   
3                1                2       1        NaN             0   
4                1                1       1        NaN             0   

   TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  CO_UF_PROVA SG_UF_PROVA  \
0                 NaN                   Brasília           53          DF   
1                 NaN                   Brasília           53          DF   
2                 NaN  Presidente Tancredo Neves           29          BA   
3                 NaN                  Cariacica           32          ES   
4                 NaN                Salinópolis           15          PA   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  \
0               0               0               0               0   
1               0               0               0               0   
2               1               1               1               1   
3               1               1               1               1   
4               0               0               0               0   

   CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  NU_NOTA_CN  NU_NOTA_CH  \
0          NaN          NaN          NaN          NaN         NaN         NaN   
1          NaN          NaN          NaN          NaN         NaN         NaN   
2       1087.0       1056.0       1066.0       1078.0       42.11       54.60   
3       1088.0       1058.0       1067.0       1077.0       49.07       38.86   
4          NaN          NaN          NaN          NaN         NaN         NaN   

   NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0         NaN         NaN          0                NaN              NaN  
1         NaN         NaN          0                NaN              NaN  
2       49.88       56.53          1                1.0            760.0  
3       35.78       41.60          1                1.0            320.0  
4         NaN         NaN          0                NaN              NaN

#Renomeando Registros (Linhas)

In [16]:
dados1["RACA"] = dados1["RACA"].replace({0:"nao_declarado", 1:"branca", 2:"preta", 3:"parda", 4:"amarela", 5:"indigena"})

In [17]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO  ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M             2   preta              1   
1  210057516120    2022     14    M             2  branca              1   
2  210057280536    2022      5    F             1   preta              1   
3  210055724397    2022      6    M             1   parda              1   
4  210055097896    2022      4    M             0   parda              1   

   TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  ESCOLA  TP_ENSINO  IN_TREINEIRO  \
0                1                2       1        NaN             0   
1                1               16       1        NaN             0   
2                1                2       1        NaN             0   
3                1                2       1        NaN             0   
4                1                1       1        NaN             0   

   TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  CO_UF_PROVA SG_UF_PROVA  \
0                 NaN                   Brasília           53          DF   
1                 NaN                   Brasília           53          DF   
2                 NaN  Presidente Tancredo Neves           29          BA   
3                 NaN                  Cariacica           32          ES   
4                 NaN                Salinópolis           15          PA   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  \
0               0               0               0               0   
1               0               0               0               0   
2               1               1               1               1   
3               1               1               1               1   
4               0               0               0               0   

   CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  NU_NOTA_CN  NU_NOTA_CH  \
0          NaN          NaN          NaN          NaN         NaN         NaN   
1          NaN          NaN          NaN          NaN         NaN         NaN   
2       1087.0       1056.0       1066.0       1078.0       42.11       54.60   
3       1088.0       1058.0       1067.0       1077.0       49.07       38.86   
4          NaN          NaN          NaN          NaN         NaN         NaN   

   NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0         NaN         NaN          0                NaN              NaN  
1         NaN         NaN          0                NaN              NaN  
2       49.88       56.53          1                1.0            760.0  
3       35.78       41.60          1                1.0            320.0  
4         NaN         NaN          0                NaN              NaN

In [18]:
dados1["ESTADO_CIVIL"] = dados1["ESTADO_CIVIL"].replace({0:"Nao informado", 1:"Solteiro(a)", 2:"Casado(a)", 3:"Divorciado(a)", 4:"Viúvo(a)"})

In [19]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M      Casado(a)   preta              1   
1  210057516120    2022     14    M      Casado(a)  branca              1   
2  210057280536    2022      5    F    Solteiro(a)   preta              1   
3  210055724397    2022      6    M    Solteiro(a)   parda              1   
4  210055097896    2022      4    M  Nao informado   parda              1   

   TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  ESCOLA  TP_ENSINO  IN_TREINEIRO  \
0                1                2       1        NaN             0   
1                1               16       1        NaN             0   
2                1                2       1        NaN             0   
3                1                2       1        NaN             0   
4                1                1       1        NaN             0   

   TP_LOCALIZACAO_ESC         NO_MUNICIPIO_PROVA  CO_UF_PROVA SG_UF_PROVA  \
0                 NaN                   Brasília           53          DF   
1                 NaN                   Brasília           53          DF   
2                 NaN  Presidente Tancredo Neves           29          BA   
3                 NaN                  Cariacica           32          ES   
4                 NaN                Salinópolis           15          PA   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  \
0               0               0               0               0   
1               0               0               0               0   
2               1               1               1               1   
3               1               1               1               1   
4               0               0               0               0   

   CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  NU_NOTA_CN  NU_NOTA_CH  \
0          NaN          NaN          NaN          NaN         NaN         NaN   
1          NaN          NaN          NaN          NaN         NaN         NaN   
2       1087.0       1056.0       1066.0       1078.0       42.11       54.60   
3       1088.0       1058.0       1067.0       1077.0       49.07       38.86   
4          NaN          NaN          NaN          NaN         NaN         NaN   

   NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0         NaN         NaN          0                NaN              NaN  
1         NaN         NaN          0                NaN              NaN  
2       49.88       56.53          1                1.0            760.0  
3       35.78       41.60          1                1.0            320.0  
4         NaN         NaN          0                NaN              NaN

In [20]:
dados1 = dados1.drop(columns=['TP_ANO_CONCLUIU','TP_ENSINO'],)

In [21]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M      Casado(a)   preta              1   
1  210057516120    2022     14    M      Casado(a)  branca              1   
2  210057280536    2022      5    F    Solteiro(a)   preta              1   
3  210055724397    2022      6    M    Solteiro(a)   parda              1   
4  210055097896    2022      4    M  Nao informado   parda              1   

   TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO  TP_LOCALIZACAO_ESC  \
0                1       1             0                 NaN   
1                1       1             0                 NaN   
2                1       1             0                 NaN   
3                1       1             0                 NaN   
4                1       1             0                 NaN   

          NO_MUNICIPIO_PROVA  CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  \
0                   Brasília           53          DF               0   
1                   Brasília           53          DF               0   
2  Presidente Tancredo Neves           29          BA               1   
3                  Cariacica           32          ES               1   
4                Salinópolis           15          PA               0   

   TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  \
0               0               0               0          NaN          NaN   
1               0               0               0          NaN          NaN   
2               1               1               1       1087.0       1056.0   
3               1               1               1       1088.0       1058.0   
4               0               0               0          NaN          NaN   

   CO_PROVA_LC  CO_PROVA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  \
0          NaN          NaN         NaN         NaN         NaN         NaN   
1          NaN          NaN         NaN         NaN         NaN         NaN   
2       1066.0       1078.0       42.11       54.60       49.88       56.53   
3       1067.0       1077.0       49.07       38.86       35.78       41.60   
4          NaN          NaN         NaN         NaN         NaN         NaN   

   TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0          0                NaN              NaN  
1          0                NaN              NaN  
2          1                1.0            760.0  
3          1                1.0            320.0  
4          0                NaN              NaN

In [22]:
dados1 = dados1.drop(columns=['TP_LOCALIZACAO_ESC',],)

In [23]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M      Casado(a)   preta              1   
1  210057516120    2022     14    M      Casado(a)  branca              1   
2  210057280536    2022      5    F    Solteiro(a)   preta              1   
3  210055724397    2022      6    M    Solteiro(a)   parda              1   
4  210055097896    2022      4    M  Nao informado   parda              1   

   TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO         NO_MUNICIPIO_PROVA  \
0                1       1             0                   Brasília   
1                1       1             0                   Brasília   
2                1       1             0  Presidente Tancredo Neves   
3                1       1             0                  Cariacica   
4                1       1             0                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0         NaN         NaN         NaN         NaN          0   
1         NaN         NaN         NaN         NaN          0   
2       42.11       54.60       49.88       56.53          1   
3       49.07       38.86       35.78       41.60          1   
4         NaN         NaN         NaN         NaN          0   

   TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0                NaN              NaN  
1                NaN              NaN  
2                1.0            760.0  
3                1.0            320.0  
4                NaN              NaN

In [24]:
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [25]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [26]:
dados1.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M      Casado(a)   preta              1   
1  210057516120    2022     14    M      Casado(a)  branca              1   
2  210057280536    2022      5    F    Solteiro(a)   preta              1   
3  210055724397    2022      6    M    Solteiro(a)   parda              1   
4  210055097896    2022      4    M  Nao informado   parda              1   

   TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO         NO_MUNICIPIO_PROVA  \
0                1  não_respondeu             0                   Brasília   
1                1  não_respondeu             0                   Brasília   
2                1  não_respondeu             0  Presidente Tancredo Neves   
3                1  não_respondeu             0                  Cariacica   
4                1  não_respondeu             0                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT TP_LINGUA  \
0         NaN         NaN         NaN         NaN    Inglês   
1         NaN         NaN         NaN         NaN    Inglês   
2       42.11       54.60       49.88       56.53  Espanhol   
3       49.07       38.86       35.78       41.60  Espanhol   
4         NaN         NaN         NaN         NaN    Inglês   

   TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0                NaN              NaN  
1                NaN              NaN  
2                1.0            760.0  
3                1.0            320.0  
4                NaN              NaN

#Análise Exploratória e Transformação dos dados

##Análise das idades

In [27]:
dados1['IDADE'].value_counts()


3     805862
2     711278
4     408115
1     303605
5     247679
11    189700
6     165982
7     123260
12    103634
8      94849
13     74162
9      73338
10     61564
14     49735
15     30066
16     17688
17      9688
18      3817
19      1505
20       578
Name: IDADE, dtype: int64

In [28]:
dados1['IDADE'].value_counts().sort_index()

1     303605
2     711278
3     805862
4     408115
5     247679
6     165982
7     123260
8      94849
9      73338
10     61564
11    189700
12    103634
13     74162
14     49735
15     30066
16     17688
17      9688
18      3817
19      1505
20       578
Name: IDADE, dtype: int64

In [29]:
# Para ver todas as linhas, primeiro transforme num Dataframe:
idade = pd.DataFrame(dados1['IDADE'].value_counts().sort_index())

In [30]:
# Configura o Pandas para observar 80 linhas
pd.set_option('display.max_rows', 80)

In [31]:
idade.head(80)

IDADE
1   303605
2   711278
3   805862
4   408115
5   247679
6   165982
7   123260
8    94849
9    73338
10   61564
11  189700
12  103634
13   74162
14   49735
15   30066
16   17688
17    9688
18    3817
19    1505
20     578

##Alguns valores estranhos e curiosos: idades menores que 12 anos e maiores de 80 anos.

In [33]:
#contagem menores de 12 anos separados por municipio
#query compara uma coluna em relação com a outra
menores_12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_PROVA'].value_counts()

In [34]:
# Inscrições com idade abaixo de 12 anos serão excluídos devido a alta probabilidade de erro.
menores_12

São Paulo              136198
Rio de Janeiro          94320
Fortaleza               64887
Brasília                58641
Belém                   49816
                        ...  
Brejinho de Nazaré         53
Santa Rosa do Purus        48
Japurá                     48
Barra do Turvo             48
Fernando de Noronha        31
Name: NO_MUNICIPIO_PROVA, Length: 1712, dtype: int64

In [35]:
dados_maiores_11 = dados1.loc[dados1.IDADE > 11]

In [36]:
dados_maiores_11.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0   210057943671    2022     14    M      Casado(a)   preta              1   
1   210057516120    2022     14    M      Casado(a)  branca              1   
31  210054534656    2022     12    M    Solteiro(a)  branca              1   
34  210056212765    2022     14    F  Divorciado(a)   parda              1   
37  210056368452    2022     12    F    Solteiro(a)   parda              1   

    TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO NO_MUNICIPIO_PROVA  \
0                 1  não_respondeu             0           Brasília   
1                 1  não_respondeu             0           Brasília   
31                1  não_respondeu             0            Erechim   
34                1  não_respondeu             0         Rio Branco   
37                1  não_respondeu             0             Manaus   

    CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0            53          DF               0               0               0   
1            53          DF               0               0               0   
31           43          RS               1               1               1   
34           12          AC               0               0               0   
37           13          AM               1               1               1   

    TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0                0          NaN          NaN          NaN          NaN   
1                0          NaN          NaN          NaN          NaN   
31               1       1085.0       1057.0       1068.0       1075.0   
34               0          NaN          NaN          NaN          NaN   
37               1       1085.0       1057.0       1068.0       1075.0   

    NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT TP_LINGUA  \
0          NaN         NaN         NaN         NaN    Inglês   
1          NaN         NaN         NaN         NaN    Inglês   
31       38.55       41.59       41.56       35.24    Inglês   
34         NaN         NaN         NaN         NaN  Espanhol   
37       50.47       55.54       57.29       47.47    Inglês   

    TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0                 NaN              NaN  
1                 NaN              NaN  
31                1.0            360.0  
34                NaN              NaN  
37                1.0            360.0

In [37]:
dados_maiores_11['IDADE'].value_counts().sort_index()

12    103634
13     74162
14     49735
15     30066
16     17688
17      9688
18      3817
19      1505
20       578
Name: IDADE, dtype: int64

**Treineiros**

In [38]:
dados_maiores_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

1    900
Name: IN_TREINEIRO, dtype: int64

In [39]:
treineiros = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 1]

In [40]:
treineiros.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO ESTADO_CIVIL      RACA  NACIONALIDADE  \
480    210055266398    2022     13    F  Solteiro(a)    branca              1   
1045   210056806257    2022     12    F    Casado(a)  indigena              1   
3226   210057588992    2022     12    M  Solteiro(a)     parda              1   
7254   210055874999    2022     12    M  Solteiro(a)     preta              1   
11354  210054480490    2022     13    F  Solteiro(a)     preta              1   

       TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO NO_MUNICIPIO_PROVA  \
480                  3  não_respondeu             1           Itaboraí   
1045                 3  não_respondeu             1     Rio de Janeiro   
3226                 3  não_respondeu             1          Fortaleza   
7254                 3  não_respondeu             1             Franca   
11354                3  não_respondeu             1            Goiânia   

       CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  \
480             33          RJ               1               1   
1045            33          RJ               0               0   
3226            23          CE               0               0   
7254            35          SP               1               1   
11354           52          GO               1               1   

       TP_PRESENCA_LC  TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  \
480                 1               1       1085.0       1057.0       1068.0   
1045                0               0          NaN          NaN          NaN   
3226                0               0          NaN          NaN          NaN   
7254                1               1       1086.0       1055.0       1065.0   
11354               1               1       1085.0       1057.0       1068.0   

       CO_PROVA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT TP_LINGUA  \
480         1075.0       37.84       54.38       53.21       40.25  Espanhol   
1045           NaN         NaN         NaN         NaN         NaN  Espanhol   
3226           NaN         NaN         NaN         NaN         NaN    Inglês   
7254        1076.0       47.38       48.51       54.37       40.14    Inglês   
11354       1075.0       40.09       52.11       45.77       37.98  Espanhol   

       TP_STATUS_REDACAO  NU_NOTA_REDACAO  
480                  1.0            540.0  
1045                 NaN              NaN  
3226                 NaN              NaN  
7254                 1.0            420.0  
11354                1.0            520.0

In [41]:
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding = 'iso-8859-1', index = False)

In [42]:
# Separação entre os treineiros e os vestibulandos
vestibulandos = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 0]

In [43]:
vestibulandos.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0   210057943671    2022     14    M      Casado(a)   preta              1   
1   210057516120    2022     14    M      Casado(a)  branca              1   
31  210054534656    2022     12    M    Solteiro(a)  branca              1   
34  210056212765    2022     14    F  Divorciado(a)   parda              1   
37  210056368452    2022     12    F    Solteiro(a)   parda              1   

    TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO NO_MUNICIPIO_PROVA  \
0                 1  não_respondeu             0           Brasília   
1                 1  não_respondeu             0           Brasília   
31                1  não_respondeu             0            Erechim   
34                1  não_respondeu             0         Rio Branco   
37                1  não_respondeu             0             Manaus   

    CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0            53          DF               0               0               0   
1            53          DF               0               0               0   
31           43          RS               1               1               1   
34           12          AC               0               0               0   
37           13          AM               1               1               1   

    TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0                0          NaN          NaN          NaN          NaN   
1                0          NaN          NaN          NaN          NaN   
31               1       1085.0       1057.0       1068.0       1075.0   
34               0          NaN          NaN          NaN          NaN   
37               1       1085.0       1057.0       1068.0       1075.0   

    NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT TP_LINGUA  \
0          NaN         NaN         NaN         NaN    Inglês   
1          NaN         NaN         NaN         NaN    Inglês   
31       38.55       41.59       41.56       35.24    Inglês   
34         NaN         NaN         NaN         NaN  Espanhol   
37       50.47       55.54       57.29       47.47    Inglês   

    TP_STATUS_REDACAO  NU_NOTA_REDACAO  
0                 NaN              NaN  
1                 NaN              NaN  
31                1.0            360.0  
34                NaN              NaN  
37                1.0            360.0

In [44]:
vestibulandos.shape

(289973, 28)

In [45]:
vestibulandos['TP_PRESENCA_CN'].value_counts()

0    151092
1    138753
2       128
Name: TP_PRESENCA_CN, dtype: int64

In [46]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

1    149363
0    140251
2       359
Name: TP_PRESENCA_CH, dtype: int64

In [47]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

1    149363
0    140251
2       359
Name: TP_PRESENCA_LC, dtype: int64

In [48]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

0    151092
1    138753
2       128
Name: TP_PRESENCA_MT, dtype: int64

RESUMO DA PRESENÇA NAS PROVAS:

Ciências da Natureza: 138753 presentes, 151092 faltaram e 128 eliminados.

Ciências Humanas: 149363 presentes, 140251 faltaram e 359 eliminados.

Linguagens e Códigos: 149363 presentes, 140251 faltaram e 359 eliminados.

Matemática: 138753 presentes, 151092 faltaram e 128 eliminados.

In [70]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

1.0    138851
2.0       524
3.0       985
4.0      2308
6.0      5777
7.0       228
8.0       421
9.0       269
Name: TP_STATUS_REDACAO, dtype: int64

##**Valores Ausentes**

In [68]:
vestibulandos['IDADE'].isnull().sum()

0

In [74]:
vestibulandos = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [75]:
vestibulandos.head()

NU_INSCRICAO  NU_ANO  IDADE SEXO   ESTADO_CIVIL    RACA  NACIONALIDADE  \
0  210057943671    2022     14    M      Casado(a)   preta              1   
1  210057516120    2022     14    M      Casado(a)  branca              1   
2  210057280536    2022      5    F    Solteiro(a)   preta              1   
3  210055724397    2022      6    M    Solteiro(a)   parda              1   
4  210055097896    2022      4    M  Nao informado   parda              1   

   TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO         NO_MUNICIPIO_PROVA  \
0                1  não_respondeu             0                   Brasília   
1                1  não_respondeu             0                   Brasília   
2                1  não_respondeu             0  Presidente Tancredo Neves   
3                1  não_respondeu             0                  Cariacica   
4                1  não_respondeu             0                Salinópolis   

   CO_UF_PROVA SG_UF_PROVA  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0           53          DF               0               0               0   
1           53          DF               0               0               0   
2           29          BA               1               1               1   
3           32          ES               1               1               1   
4           15          PA               0               0               0   

   TP_PRESENCA_MT  CO_PROVA_CN  CO_PROVA_CH  CO_PROVA_LC  CO_PROVA_MT  \
0               0          NaN          NaN          NaN          NaN   
1               0          NaN          NaN          NaN          NaN   
2               1       1087.0       1056.0       1066.0       1078.0   
3               1       1088.0       1058.0       1067.0       1077.0   
4               0          NaN          NaN          NaN          NaN   

   NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  \
0      NaN      NaN      NaN      NaN    Inglês                NaN   
1      NaN      NaN      NaN      NaN    Inglês                NaN   
2    42.11    54.60    49.88    56.53  Espanhol                1.0   
3    49.07    38.86    35.78    41.60  Espanhol                1.0   
4      NaN      NaN      NaN      NaN    Inglês                NaN   

   NU_NOTA_REDACAO  
0              NaN  
1              NaN  
2            760.0  
3            320.0  
4              NaN

In [76]:
provas = ['NU_NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [77]:
# Registros NAN nas notas
vestibulandos[provas].isnull().sum()

NU_NOTA_REDACAO     982663
NOTA_CN            1120710
NOTA_MT            1120710
NOTA_LC             982663
NOTA_CH             982663
dtype: int64

In [78]:
vestibulandos_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
vestibulandos_classificados = vestibulandos_classificados.loc[vestibulandos_classificados.TP_PRESENCA_MT == 1]

In [79]:
vestibulandos_classificados['TP_PRESENCA_MT'].value_counts()

1    2344823
Name: TP_PRESENCA_MT, dtype: int64

In [80]:
vestibulandos_classificados['TP_PRESENCA_CH'].value_counts()

1    2344823
Name: TP_PRESENCA_CH, dtype: int64

In [81]:
vestibulandos_classificados['TP_PRESENCA_CN'].value_counts()

1    2344823
Name: TP_PRESENCA_CN, dtype: int64

In [82]:
vestibulandos_classificados['TP_PRESENCA_LC'].value_counts()

1    2344823
Name: TP_PRESENCA_LC, dtype: int64

In [83]:
vestibulandos_classificados[provas].isnull().sum()

NU_NOTA_REDACAO    0
NOTA_CN            0
NOTA_MT            0
NOTA_LC            0
NOTA_CH            0
dtype: int64

In [86]:
vestibulandos_classificados = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO',})

** Notas zeros na Redação**

In [88]:
from collections import Counter
dict(Counter(vestibulandos_classificados['NOTA_REDACAO']))

{nan: 1,
 nan: 1,
 760.0: 79317,
 320.0: 20957,
 nan: 1,
 nan: 1,
 nan: 1,
 440.0: 41878,
 nan: 1,
 360.0: 42033,
 nan: 1,
 940.0: 42337,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 640.0: 115979,
 860.0: 63254,
 nan: 1,
 0.0: 129827,
 nan: 1,
 nan: 1,
 520.0: 97525,
 nan: 1,
 nan: 1,
 300.0: 10685,
 nan: 1,
 nan: 1,
 nan: 1,
 540.0: 103907,
 420.0: 25950,
 460.0: 43853,
 560.0: 140869,
 720.0: 86730,
 600.0: 143777,
 580.0: 131221,
 840.0: 74675,
 480.0: 65318,
 900.0: 64089,
 820.0: 62562,
 740.0: 69311,
 660.0: 88881,
 nan: 1,
 500.0: 64209,
 nan: 1,
 nan: 1,
 780.0: 63691,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 340.0: 25411,
 nan: 1,
 nan: 1,
 nan: 1,
 700.0: 77423,
 nan: 1,
 nan: 1,
 nan: 1,
 880.0: 74374,
 nan: 1,
 620.0: 112907,
 nan: 1,
 920.0: 63243,
 nan: 1,
 800.0: 75100,
 nan: 1,
 nan: 1,
 380.0: 21588,
 nan: 1,
 nan: 1,
 680.0: 98096,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan: 1,
 nan:

In [89]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

0.0    129827
Name: NOTA_REDACAO, dtype: int64

NOTAS ZEROS EM REDAÇÃO:

REDAÇÃO (final): 129827 notas zeros devido a não cumprimento das regras ou tirou zero em todos os cinco componentes.